src/IK/include/BipedalLocomotion/IK/QPInverseKinematics.h

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file QPInverseKinematics.h
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_IK_QP_INVERSE_KINEMATICS_H
+#define BIPEDAL_LOCOMOTION_IK_QP_INVERSE_KINEMATICS_H
+
+#include <memory>
+#include <vector>
+#include <string>
+
+
+#include <Eigen/Dense>
+
+#include <BipedalLocomotion/IK/InverseKinematics.h>
+#include <BipedalLocomotion/IK/Task.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+namespace BipedalLocomotion
+{
+
+namespace IK
+{
+
+/**
+ * QPInverseKinematics is a concrete class and implements an integration base inverse kinematics.
+ * The inverse kinematics is here implemented as Quadratic Programming (QP) problem. The user should
+ * set the desired task with the method QPInverseKinematics::addTask. Each task has a given
+ * priority. Currently we support only priority equal to 0 or 1. If the task priority is set to 0
+ * the task will be considered as hard task, thus treated as an equality constraint. If the priority
+ * is equal to 1 the task will be embedded in the cost function. The class is also able to treat
+ * inequality constraints.
+ * A possible usage of the IK can be found in "Romualdi et al. A Benchmarking of DCM Based
+ * Architectures for Position and Velocity Controlled Walking of Humanoid Robots"
+ * https://doi.org/10.1109/HUMANOIDS.2018.8625025
+ */
+class QPInverseKinematics : public InverseKinematics
+{
+    /**
+     * Private implementation
+     */
+    struct Impl;
+    std::unique_ptr<Impl> m_pimpl;
+
+public:
+
+    /**
+     * Constructor.
+     */
+    QPInverseKinematics();
+
+    /**
+     * Destructor.
+     */
+    ~QPInverseKinematics();
+
+    /**
+     * Add a task in the inverse kinematics
+     * @param task pointer to a given task
+     * @param priority is the priority of the task. The lower the number the higher the priority.
+     * @param weight weight associated to the task.
+     * @param isInequality true if the task should be treated as inequality constraint
+     * @note currently we support only task with priority 0 or 1. If the priority is set to 0 the
+     * task will be considered as equality constraint. This the weight is not required.
```

Comment:
So, if ``priority`` is 0, the task will be considered an equality independently of ``isInequality``? 
Also
>```
>This the weight is not required.
>```
seems to have a typo.